### PR TITLE
Saved model

### PR DESCRIPTION
**Pull request by dtrawins**

### CODE REVIEW

File: src/modelinstance.hpp [hunk 1]
```diff
@@ -367,6 +367,14 @@ class ModelInstance {
         return path;
     }
 
+    /**
+         * @brief Gets model files' paths
+         *
+         * @return vector of paths
+         */
+    const std::vector<std::string> getModelFiles() const {
```
Comment (atobiszei) [inline src/modelinstance.hpp:375]: Avoid copy
```suggestion
    const std::vector<std::string>& getModelFiles() const {
```

File: src/test/modelinstance_test.cpp [hunk 2]
```diff
@@ -360,6 +368,108 @@ TEST_F(TestLoadModel, CheckIfNonExistingBinFileReturnsFileInvalid) {
     EXPECT_EQ(status, ovms::StatusCode::FILE_INVALID) << status.string();
 }
 
+TEST_F(TestLoadModel, CheckSavedModelHandling) {
```
Comment (atobiszei) [inline src/test/modelinstance_test.cpp:371]: Reuse test fixture:
TestWithTempDir() from test_utils.hpp

You need to inherit from that fixture, and then the temporary files are cleaned up after execution. You can check example usage in StressPipelineConfigChanges fixture.

Comment (atobiszei) [reply src/test/modelinstance_test.cpp:371]: Additionally add test for loading such model and verifying at least the model state == AVAILABLE.

**Review by michalkulakowski — APPROVED** [review]

File: src/test/modelinstance_test.cpp [hunk 3]
```diff
@@ -360,6 +369,108 @@ TEST_F(TestLoadModel, CheckIfNonExistingBinFileReturnsFileInvalid) {
...
+        false,      // is stateful
+        false,      // idle sequence cleanup enabled
+        false,      // low latency transformation enabled
+        500,        // stateful sequence max number,
+        "",         // cache dir
+        version,    // version
+        modelPath,  // local path
+    };
+    auto status = modelInstance.loadModel(config);
+    auto model_files = modelInstance.getModelFiles();
```
Comment (dkalinowski) [inline src/test/modelinstance_test.cpp:435]: Assert number of model files (should be 1?). We can avoid segfaults in lin 437/403/471 in case size is 0

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**